File: wiz-event-mall-java/src/main/java/com/example/wizeventmalljava/api/event/dto/EventParticipateResponse.java
package com.example.wizeventmalljava.api.event.dto;

import com.example.wizeventmalljava.domain.event.entity.EventWinner;

public record EventParticipateResponse(
	boolean isWinner
) {
	public static EventParticipateResponse from(EventWinner eventWinner) {
		return new EventParticipateResponse(eventWinner != null);
	}
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/api/event/dto/EventParticipateResponse.kt
package com.example.wizeventmall.api.event.dto

import com.example.wizeventmall.domain.event.entity.EventWinner

data class EventParticipateResponse(
    val isWinner: Boolean
) {

    companion object {
        fun from(result: EventWinner?) = EventParticipateResponse(
            isWinner = result != null
        )
    }
}